Router/Router.swift

```diff
@@ -0,0 +1,207 @@
+public protocol RouteTarget {
+    var baseURL: NSURL { get }
+    var endpoint: Route.Endpoint { get }
+    var parameters: Route.Parameters { get }
+}
+
+public protocol MappableRouteTarget {
+    typealias Model: Mappable
+}
+
+public protocol RouteHeaders {
+    var headers: [String: AnyObject] { get }
+}
+
+/// Protocol to define the opaque type returned from a request
+public protocol Cancellable {
+    func cancel()
+}
+
+/// Internal token that can be used to cancel requests
+struct CancellableToken: Cancellable {
+    let cancelAction: () -> ()
+    
+    func cancel() {
+        cancelAction()
+    }
+}
+
+
+public final class Route {
+    
+    /// Represents an HTTP method.
+    public enum Endpoint {
+        case GET(String)
+        case POST(String)
+        case PUT(String)
+        case DELETE(String)
+        case OPTIONS(String)
+        case HEAD(String)
+        case PATCH(String)
+        case TRACE(String)
+        case CONNECT(String)
+        
+        func method() -> Alamofire.Method {
+            switch self {
+            case .GET:
+                return .GET
+            case .POST:
+                return .POST
+            case .PUT:
+                return .PUT
+            case .DELETE:
+                return .DELETE
+            case .HEAD:
+                return .HEAD
+            case .OPTIONS:
+                return .OPTIONS
+            case PATCH:
+                return .PATCH
+            case TRACE:
+                return .TRACE
+            case .CONNECT:
+                return .CONNECT
+            }
+        }
+        
+        func path() -> String {
+            switch self {
+            case .GET(let path):
+                return path
+            case .POST(let path):
+                return path
+            case .PUT(let path):
+                return path
+            case .DELETE(let path):
+                return path
+            case .HEAD(let path):
+                return path
+            case .OPTIONS(let path):
+                return path
+            case PATCH(let path):
+                return path
+            case TRACE(let path):
+                return path
+            case .CONNECT(let path):
+                return path
+            }
+        }
+    }
+    
+    public enum Parameters {
+        case None
+        case JSONString(String)
+        case JSONArray([AnyObject])
+        case JSON([String:AnyObject])
+        case URL([String:AnyObject])
+    }
+    
+}
+
+public class Router<T: RouteTarget> {
+
+    public func request(target: T, completion: RouterCompletion) -> Cancellable {
+        var headers: [String: AnyObject] = [:]
+        
+        var url = target.baseURL.URLByAppendingPathComponent(target.endpoint.path())
+
+        var request: NSMutableURLRequest = NSMutableURLRequest(URL: url)
+        request.HTTPMethod = target.endpoint.method().rawValue
+        
+        request.allHTTPHeaderFields = headers
```

Comment:
Why not `request.allHTTPHeaderFields = [:]` and remove `var headers: [String: AnyObject] = [:] ` ?